### PR TITLE
Address some demo feedback, switch to using web components for video tiles

### PR DESCRIPTION
**Pull request by hensmi-amazon**
**Issue #:** 

**Description of changes:** Followup to https://github.com/aws/amazon-chime-sdk-js/pull/1594 , addressing some additional comments, and switching to using [Web Components](https://www.google.com/search?channel=cus2&client=firefox-b-1-d&q=web+components) for the video tile portion (seems like theres some gaps in definitive 'how to do this or that' with Web Components so pardon anything egregious; e.g. if i want to use a template for a custom element, should i add it as plaintext in JS? or stuff in .html file?  I see it both ways online).

**Testing**
Smoke test 1-3 person calls.  All enabling video and disabling multiple times.  Hover over attendee ID shows.  Dropdown menu works as before.  Stats hover enable works as before (I want to move this to drop down TBH).  Active speaker layout works.  Content share video works.

1. Have you successfully run `npm run build:release` locally? y
2. How did you test these changes? see above
3. Can these changes be tested using the demo application? If yes, which demo application can be used to test it? meetingdemov2
4. Do you add, modify, or delete public API definitions? If yes, has that been reviewed and approved? n
5. Do you change the wire protocol, e.g. the request method? If yes, has that been reviewed and approved? n

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: demos/browser/app/meetingV2/meetingV2.ts [hunk 1]
```diff
@@ -139,6 +139,142 @@ class DemoTileOrganizer {
   }
 }
 
+class DemoVideoTile extends HTMLElement {
```
Comment (hensmi-amazon) [inline demos/browser/app/meetingV2/meetingV2.ts:142]: [I don't know why we keep everything in a 3.5k line file and at this point I'm too afraid to ask](https://i.kym-cdn.com/entries/icons/original/000/016/724/prattypratprat.PNG)

Comment (ltrung) [reply demos/browser/app/meetingV2/meetingV2.ts:142]: We can move it out though right similar to video filter processor

File: demos/browser/app/meetingV2/meetingV2.ts [hunk 2]
```diff
@@ -3501,9 +3538,24 @@ export class DemoMeetingApp
     return tiles;
   }
 
-  setUpVideoTileElementResizer(): void {
+  setupVideoTiles(): void {
+    const tileArea = document.getElementById(`tile-area`);
     for (let i = 0; i <= DemoTileOrganizer.MAX_TILES; i++) {
-      const videoElem = document.getElementById(`video-${i}`) as HTMLVideoElement;
+      let tile = document.createElement('video-tile') as DemoVideoTile
```
Comment (ltrung) [inline demos/browser/app/meetingV2/meetingV2.ts:3544]: nit: const instead of let?

**Comment by ltrung** [comment]
Thanks for doing this :D

File: demos/browser/app/meetingV2/meetingV2.ts (hunk 2 above)
Comment (hensmi-amazon) [reply demos/browser/app/meetingV2/meetingV2.ts:3544]: Need to modify a couple lines below.

File: demos/browser/app/meetingV2/meetingV2.ts [hunk 3]
```diff
@@ -139,6 +139,142 @@ class DemoTileOrganizer {
...
 }
 
+class DemoVideoTile extends HTMLElement {
+  public set tileIndex(tileIndex: number) {
+    // Update IDs for integration tests which need them
+    this.querySelector('.video-tile-nameplate').id = `nameplate-${tileIndex}`
+    this.querySelector('.video-tile-video').id = `video-${tileIndex}`
+  }
+
+  public set showConfigDropdown(v : boolean) {
```
Comment (devalevenkatesh) [inline demos/browser/app/meetingV2/meetingV2.ts:149]: Sorry did not understand `v` here, can we have a descriptive name?

File: demos/browser/app/meetingV2/meetingV2.ts (hunk 1 above)
Comment (ltrung) [reply demos/browser/app/meetingV2/meetingV2.ts:142]: We could move it to a separate one like video filter right?

File: demos/browser/app/meetingV2/meetingV2.ts (hunk 2 above)
Comment (ltrung) [reply demos/browser/app/meetingV2/meetingV2.ts:3544]: You just set properties right? I think you can modify properties for const object :D

File: demos/browser/app/meetingV2/meetingV2.ts (hunk 1 above)
Comment (hensmi-amazon) [reply demos/browser/app/meetingV2/meetingV2.ts:142]: I'll save it for another PR.  It feels a bit strange to just have this in its own file given all the other tile logic is either in this file or the CSS :P.

**Review by ltrung — APPROVED** [review]

File: demos/browser/app/meetingV2/meetingV2.ts [hunk 4]
```diff
@@ -139,6 +139,142 @@ class DemoTileOrganizer {
...
+    }
+  }
+
+  public set featured(featured: boolean) {
+    if (featured) {
+      this.classList.add(`featured`);
+    } else {
+      this.classList.remove(`featured`);
+    }
+  }
```
Comment (richnew10) [inline demos/browser/app/meetingV2/meetingV2.ts:176]: Standardize on straight single quotes for strings that don't actually use templating.

File: demos/browser/app/meetingV2/meetingV2.ts [hunk 5]
```diff
@@ -139,6 +139,142 @@ class DemoTileOrganizer {
...
+            `${metricName}`
+          ) as HTMLTableRowElement;
+          const row = rowElement ? rowElement : statsInfoTable.insertRow(-1);
+          if (!rowElement) {
+            row.setAttribute('id', `${metricName}`);
+            cell = row.insertCell(-1);
+            cell.innerHTML = keyStatstoShow[metricName];
+          }
+            cell = row.insertCell(-1);
+            cell.innerHTML = `${value}`;
```
Comment (richnew10) [inline demos/browser/app/meetingV2/meetingV2.ts:230]: Indenting.

File: demos/browser/app/meetingV2/meetingV2.ts [hunk 6]
```diff
@@ -139,6 +139,142 @@ class DemoTileOrganizer {
...
+    const nameplateElement = this.querySelector('.video-tile-nameplate')
+    console.log(`setting nameplate to ${nameplate}`)
+    nameplateElement.innerHTML = nameplate
+  }
+
+  public set attendeeId(attendeeId: string) {
+    const attendeeIdElement = this.querySelector('.video-tile-attendee-id')
+    console.log(`setting attendeeId to ${attendeeId}`)
+    attendeeIdElement.innerHTML = attendeeId
+  }
```
Comment (richnew10) [inline demos/browser/app/meetingV2/meetingV2.ts:250]: Semicolons, throughout.

**Review by richnew10 — CHANGES_REQUESTED** [review]

File: demos/browser/app/meetingV2/meetingV2.ts (hunk 5 above)
Comment (hensmi-amazon) [reply demos/browser/app/meetingV2/meetingV2.ts]: Ergh, need to get the auto formatter/linter to run on demo as well.  todo....

**Review by ltrung — APPROVED** [review]

**Review by richnew10 — APPROVED** [review]

**Pull request merged**